vignettes/loo2-moment-matching.Rmd

```diff
@@ -0,0 +1,309 @@
+Pareto $k$ diagnostics for each observation. Here, we present a method for
+quickly improving
+the accuracy when the Pareto diagnostics indicate problems. This is done
+by performing some additional computations using the existing posterior sample.
+If successful, this will decrease the Pareto $k$ values, making
+the model assessment more reliable. __loo__ also stores
+the original Pareto $k$ values with the name `influence_pareto_k` which are
+not changed. They can be used as a diagnostic of how much
+each observation influences the posterior distribution.
+The methodology presented is based on the paper
+
+* Paananen, T., Piironen, J., Buerkner, P.-C., Vehtari, A. (2019).
+Pushing the Limits of Importance Sampling through Iterative Moment Matching.
+[arXiv preprint arXiv:1906.08850](http://arxiv.org/abs/1906.08850).
+
+More information about the Pareto $k$ diagnostics is given in the following
+papers
+
+* Vehtari, A., Gelman, A., and Gabry, J. (2017). Practical Bayesian model evaluation using leave-one-out cross-validation and WAIC. _Statistics and Computing_. 27(5), 1413--1432. \doi:10.1007/s11222-016-9696-4. Links: [published](http://link.springer.com/article/10.1007\%2Fs11222-016-9696-4) | [arXiv preprint](http://arxiv.org/abs/1507.04544).
+
+* Vehtari, A., Simpson, D., Gelman, A., Yao, Y., and Gabry, J. (2019). Pareto smoothed importance sampling. [arXiv preprint arXiv:1507.02646](http://arxiv.org/abs/1507.02646).
+
+# Example: Eradication of Roaches
+
+We will use the same example as in the vignette
+[_Using the loo package (version >= 2.0.0)_](http://mc-stan.org/loo/articles/loo2-example.html).
+See the demo for a description of the problem and data. We will use the same Poisson
+regression model as in the case study.
+
+## Coding the Stan model
+
+Here is the Stan code for fitting the Poisson regression model, which 
+we will use for modelling the number of roaches.
+
+```{r, stancode}
+stancode = "
+data {
+  int<lower=1> K;
+  int<lower=1> N;
+  matrix[N,K] x;
+  int y[N];
+  vector[N] offset;
+
+  real beta_prior_scale;
+  real alpha_prior_scale;
+}
+parameters {
+  vector[K] beta;
+  real intercept;
+}
+model {
+  y ~ poisson(exp(x * beta + intercept + offset));
+  beta ~ normal(0,beta_prior_scale);
+  intercept ~ normal(0,alpha_prior_scale);
+}
+generated quantities {
+  vector[N] log_lik;
+  for (n in 1:N)
+    log_lik[n] = poisson_lpmf(y[n] | exp(x[n] * beta + intercept + offset[n]));
+}
+"
+```
+
+Following the usual approach recommended in 
+[_Writing Stan programs for use with the loo package_](http://mc-stan.org/loo/articles/loo2-with-rstan.html), 
+we compute the log-likelihood for each observation in the 
+`generated quantities` block of the Stan program.
+
+
+## Setup
+
+In addition to __loo__, we load the __rstan__ package for fitting the model,
+and the __rstanarm__ package for the data.
+
+```{r, setup, message=FALSE}
+library("rstan")
+library("loo")
+library("rstanarm")
```

Comment:
If we just need the data from rstanarm, we shouldn't load the package and instead load the data as `data(roaches, package = "rstanarm")`